scripts/eslint/rules/argument-trivia.cjs

```diff
@@ -0,0 +1,191 @@
+const { AST_NODE_TYPES, TSESTree, ESLintUtils } = require("@typescript-eslint/utils");
+const { createRule } = require("./utils.cjs");
+const ts = require("typescript");
+
+const unset = Symbol();
+/**
+ * @template T
+ * @param {() => T} fn
+ * @returns {() => T}
+ */
+function memoize(fn) {
+    /** @type {T | unset} */
+    let value = unset;
+    return () => {
+        if (value === unset) {
+            value = fn();
+        }
+        return value;
+    };
+}
+
+
+module.exports = createRule({
+    name: "argument-trivia",
+    meta: {
+        docs: {
+            description: ``,
+            recommended: "error",
+        },
+        messages: {
+            argumentTriviaArgumentError: `Tag argument with parameter name`,
+            argumentTriviaArgumentSpaceError: `There should be 1 space between an argument and its comment`,
+            argumentTriviaArgumentNameError: `Argument name "{{ got }}" does not match expected name "{{ want }}"`,
+        },
+        schema: [],
+        type: "problem",
+        fixable: "code",
+    },
+    defaultOptions: [],
+
+    create(context) {
+        const sourceCode = context.getSourceCode();
+        const sourceCodeText = sourceCode.getText();
+
+        /** @type {(name: string) => boolean} */
+        const isSetOrAssert = (name) => name.startsWith("set") || name.startsWith("assert");
+        /** @type {(node: TSESTree.Node) => boolean} */
+        const isTrivia = (node) => {
+            if (node.type === AST_NODE_TYPES.Identifier) {
+                return node.name === "undefined";
+            }
+
+            if (node.type === AST_NODE_TYPES.Literal) {
+                // eslint-disable-next-line no-null/no-null
+                return node.value === null || node.value === true || node.value === false;
+            }
+
+            return false;
+        };
+
+        /** @type {(node: TSESTree.CallExpression | TSESTree.NewExpression) => boolean} */
+        const shouldIgnoreCalledExpression = (node) => {
+            if (node.callee && node.callee.type === AST_NODE_TYPES.MemberExpression) {
+                const methodName = node.callee.property.type === AST_NODE_TYPES.Identifier
+                    ? node.callee.property.name
+                    : "";
+
+                if (isSetOrAssert(methodName)) {
+                    return true;
+                }
+
+                return ["apply", "call", "equal", "fail", "isTrue", "output", "stringify", "push"].indexOf(methodName) >= 0;

```

Comment:
You probably dont' want to create a new array for every call/`new`. Maybe hoist it our or

```suggestion
                switch (methodName) {
                    case "apply":
                    case "call":
                    case "equal":
                    case "fail":
                    case "isTrue":
                    case "output":
                    case "stringify":
                    case "push":
                        return true;
                }
                
                return false;
```

Reply:
Git doesn't show it, but this all is existing code, it's just not showing as such because I renamed the file but it has enough code to be seen as different.

Anyway, yep, a switch is definitely a better idea.